.maintain/chaostest/src/k8s/deployment.js

```diff
@@ -0,0 +1,165 @@
+const k8s = require('@kubernetes/client-node')
+
+// load k8s
+const kc = new k8s.KubeConfig()
+kc.loadFromDefault()
+
+// load k8s Apis
+const k8sAppApi = kc.makeApiClient(k8s.AppsV1Api)
+const k8sCoreApi = kc.makeApiClient(k8s.CoreV1Api)
+
+const createNameSpace = async namespace => {
+  const namespaceJson = {
+    apiVersion: 'v1',
+    kind: 'Namespace',
+    metadata: {
+      name: namespace,
+    },
+  }
+
+  try {
+    await k8sCoreApi.readNamespace(namespace)  // if namespace is available, do not create here
+  } catch (error) {
+    await k8sCoreApi.createNamespace(namespaceJson)
+  }
+}
+
+const createService = async (port, namespace, serviceName) => {
+  const service = {
+    apiVersion: 'v1',
+    kind: 'Service',
+    metadata: {
+        name: serviceName
+    },
+    labels: {
+        app: 'substrate-node'
+    },
+    spec: {
+        type: 'NodePort',
+        selector: {
+            app: 'substrate-node'
+        },
+        ports: [{
+           name: 'http',
+           port: 9933,
+           targetPort: 9933,
+           nodePort: port,
+           protocol: 'TCP'
+        }]
+    }
+}
+
+    // if service creation failed(ex: port allocated, existed), just delete and recreate a new one
+    try {
+        await k8sCoreApi.createNamespacedService(namespace, service)
+    } catch (err) {
+        console.log(err)
+        await k8sCoreApi.deleteNamespacedService(serviceName, namespace)
+        await k8sCoreApi.createNamespacedService(namespace, service)
+    }
+}
+
+const createDeployment = async (image, namespace, deploymentName) => {
+    const label = 'substrate-node' + process.env.CI_COMMIT_SHORT_SHA
+    const deployment = {
+        apiVersion: 'apps/v1',
+        kind: 'Deployment',
+        metadata: {
+            name: deploymentName,
+        },
+        labels: {
+            app: label,
+        },
+        spec: {
+            selector: {
+                matchLabels: {
+                    app: label,
+                },
+            },
+            template: {
+                metadata: {
+                    labels: {
+                        app: label,
+                    },
+                },
+                spec: {
+                    containers: [{
+                        name: label,
+                        image: image,
+                        ports: [{containerPort: 9933}],
+                        args: ['--dev', '--rpc-external', '--ws-external'],
+                    }],
+                },
+            },
+        },
+    }
+
+    // if deployment creation failed due to previous deployment not cleaned up, just delete and recreate a new one
```

Comment:
I suggested above to use a separate namespace for each test run. In that case this would not be necessary.